src/robot/running/context.py

```diff
@@ -126,14 +126,22 @@ def in_teardown(self):
     def variables(self):
         return self.namespace.variables
 
-    @property
-    def continue_on_failure(self):
-        parents = ([self.test] if self.test else []) + self.user_keywords
+    def continue_on_failure(self, default=False):
+        if self.in_teardown:
+            parents = self.user_keywords
+        else:
+            parents = ([self.test] if self.test else []) + self.user_keywords
         if not parents:
+            return default
+        if 'robot:stop-on-failure' in parents[-1].tags:
             return False
         if 'robot:continue-on-failure' in parents[-1].tags:
             return True
-        return any('robot:recursive-continue-on-failure' in p.tags for p in parents)
+        if any('robot:recursive-stop-on-failure' in p.tags for p in parents):
+            return False
+        if any('robot:recursive-continue-on-failure' in p.tags for p in parents):
+            return True
```

Comment:
Doesn't this work wrong if a keyword enables recursive stop-on-failure mode and then a keyword it uses enables recursive continue-on-failure mode? Shouldn't the latter, i.e. the closest parent, then have precedence? Changing the algorithm as follows ought to fix that:
```python
for parent in reversed(parents):
    if 'robot:recursive-continue-on-failure' in parent.tags:
        return True
    if 'robot:recursive-stop-on-failure' in parent.tags:
        return False
```
The above algorithm would also have a benefit that parents weren't traversed twice so it ought to be a bit faster.

Looking at tags twice with each parent is still likely to down handling failures. There were some reports related to performance degradation when we enabled controlling continue-on-failure with tags. That mainly shows if you have a lot of failures (e.g. a loop or Wait Until Keyword Succeeds) and that ought to be pretty rare. Should do some performance testing after this PR is merged and see could the situation be enhanced. Possibly looking at tags could be made faster in cases like this where the value isn't a pattern.

Reply:
One thing to decide is how to handle situation where a test or a keyword has both more than one of the tags controlling the continue-on-failure mode. It should be an error, but us detecting it would require adding additional checks and I'm not sure are they worth it. Just leaving the situation as it is could be fine. I wouldn't call it a backwards incompatible change if we made it stricter later. Linters could also easily check that.

Reply:
Hmm..  I figured the feature was designed to STOP continue-on-failure mode, so I felt that `recursive-stop-on-failure` in **ANY** parent would propagate to the bottom of the stack, no matter what. The only way to overwrite it would be a specific `robot:continue-on-failure`, but not a recursive-continue-on-failure..  
This explicit preference (stop overriding continue) would also handle the case of both tags being present in a test case (stop is checked first and hence takes precedence).. 
Maybe we should discuss on slack?

Reply:
I now implemented your suggested algorithm (`for parent in reversed(parents)`, etc.), however first checking stop, then continue.. adding test cases for this one now..